### ISSUE TITLE
Generic/UpperCaseConstantName: false positives for namespaced function called define()

### ISSUE THREAD

**Issue by jrfnl** [PHPCSStandards/PHP_CodeSniffer#733]
## Describe the bug

Inspired by #665

When a codebase contains a namespaced function called `define`, the sniff may throw false positives.

#### File 1:
```php
namespace Foo;

function define($name, $value) {
    // Do something.
}

// These are calls to the namespaced function, not the global one.
define('name', 'value');
namespace\define('name', 'value');

// This is a call to another namespaced function `Foo\Sub\define()`, not the global one.
Sub\define('name', 'value');

// This is a call to yet another namespaced function `My\Other\NS\define()`, not the global one.
\My\Other\NS\define('name', 'value');
```

#### File 2:
```php
namespace Foo;

use function Bar\define;

// This is a call to the imported namespaced function, not the global one.
define('name', 'value');
```

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php --standard=generic --sniffs=generic.namingconventions.uppercaseconstantname`
3. See error message displayed
```text
# File 1:
---------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
---------------------------------------------------------------------------------------------------------------------------------------------------
 10 | ERROR | Constants must be uppercase; expected 'NAME' but found 'name' (Generic.NamingConventions.UpperCaseConstantName.ConstantNotUpperCase)
 11 | ERROR | Constants must be uppercase; expected 'NAME' but found 'name' (Generic.NamingConventions.UpperCaseConstantName.ConstantNotUpperCase)
 14 | ERROR | Constants must be uppercase; expected 'NAME' but found 'name' (Generic.NamingConventions.UpperCaseConstantName.ConstantNotUpperCase)
 17 | ERROR | Constants must be uppercase; expected 'NAME' but found 'name' (Generic.NamingConventions.UpperCaseConstantName.ConstantNotUpperCase)
---------------------------------------------------------------------------------------------------------------------------------------------------

# File 2:
--------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------------------------------------------
 8 | ERROR | Constants must be uppercase; expected 'NAME' but found 'name' (Generic.NamingConventions.UpperCaseConstantName.ConstantNotUpperCase)
--------------------------------------------------------------------------------------------------------------------------------------------------
```

## Expected behavior

No errors.

## Versions (please complete the following information)

|                                       |                                                                                                           |
|--------------------------|----------------------------------------------------------------------  |
| Operating System           | _not relevant_                                                      |
| PHP version                    | _not relevant_                                                                                     |
| PHP_CodeSniffer version | `master`                                                                            |
| Standard                         | `Generic`                                                       |
| Install type                      | _not relevant_ |

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.
